File: electron/renderer/src/services/liveReport.ts
import { fetchJsonWithAuth } from './http';
import { apiCall } from '../utils/error-handler';
import type { ReviewData, ReportArtifacts } from '../types/report';
import type {
  StartLiveReportRequest,
  StartLiveReportResponse,
  LiveReportStatusResponse,
  GenerateLiveReportResponse,
} from '../types/api-types';

// 导出类型供其他模块使用
export type { ReviewData, ReportArtifacts } from '../types/report';

export const startLiveReport = async (
  liveUrl: string,
  segmentMinutes = 30
): Promise<StartLiveReportResponse> => {
  return apiCall(
    () => fetchJsonWithAuth('main', '/api/report/live/start', {
      method: 'POST',
      body: JSON.stringify({ live_url: liveUrl, segment_minutes: segmentMinutes } satisfies StartLiveReportRequest),
    }),
    '启动直播报告'
  );
};

export const stopLiveReport = async () => {
  return apiCall(
    () => fetchJsonWithAuth('main', '/api/report/live/stop', { method: 'POST' }),
    '停止直播报告'
  );
};

export const pauseLiveReport = async () => {
  return apiCall(
    () => fetchJsonWithAuth('main', '/api/report/live/pause', { method: 'POST' }),
    '暂停直播报告'
  );
};

export const resumeLiveReport = async () => {
  return apiCall(
    () => fetchJsonWithAuth('main', '/api/report/live/resume', { method: 'POST' }),
    '恢复直播报告'
  );
};

export const getResumableSession = async () => {
  return apiCall(
    () => fetchJsonWithAuth('main', '/api/report/live/resumable'),
    '获取可恢复会话'
  );
};

export const getLiveReportStatus = async (): Promise<LiveReportStatusResponse> => {
  return apiCall(
    () => fetchJsonWithAuth('main', '/api/report/live/status'),
    '获取直播报告状态'
  );
};

export const generateLiveReport = async (): Promise<GenerateLiveReportResponse> => {
  return apiCall(
    () => fetchJsonWithAuth('main', '/api/report/live/generate', { method: 'POST' }),
    '生成直播报告'
  );
};

File: server/local/routers/live_report.py
# -*- coding: utf-8 -*-
"""
直播报告路由
提供直播会话报告生成、历史查询等功能
"""

import logging
from typing import Dict, Any
from fastapi import APIRouter, HTTPException, Query

from ..schemas.live_report import StartLiveReportRequest

router = APIRouter(prefix="/api/report/live", tags=["直播报告"])
logger = logging.getLogger(__name__)


@router.post("/start")
async def start_live_report(request: StartLiveReportRequest) -> Dict[str, Any]:
    """开始直播报告"""
    try:
        logger.info(f"开始直播报告: {request.live_url}")
        # TODO: 实现实际的报告生成逻辑
        return {
            "success": True,
            "message": "直播报告已开始",
            "session_id": None
        }
    except Exception as e:
        logger.error(f"开始直播报告失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/stop")
async def stop_live_report() -> Dict[str, Any]:
    """停止直播报告"""
    try:
        logger.info("停止直播报告")
        return {
            "success": True,
            "message": "直播报告已停止"
        }
    except Exception as e:
        logger.error(f"停止直播报告失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/pause")
async def pause_live_report() -> Dict[str, Any]:
    """暂停直播报告"""
    try:
        logger.info("暂停直播报告")
        return {
            "success": True,
            "message": "直播报告已暂停"
        }
    except Exception as e:
        logger.error(f"暂停直播报告失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/resume")
async def resume_live_report() -> Dict[str, Any]:
    """恢复直播报告"""
    try:
        logger.info("恢复直播报告")
        return {
            "success": True,
            "message": "直播报告已恢复"
        }
    except Exception as e:
        logger.error(f"恢复直播报告失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/status")
async def get_live_report_status() -> Dict[str, Any]:
    """获取当前直播报告状态"""
    try:
        return {
            "success": True,
            "data": {
                "is_running": False,
                "is_generating": False,
                "current_session_id": None,
                "last_report_time": None
            }
        }
    except Exception as e:
        logger.error(f"获取直播报告状态失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/resumable")
async def get_resumable_sessions() -> Dict[str, Any]:
    """获取可恢复的直播会话"""
    try:
        return {
            "success": True,
            "data": {
                "sessions": []
            }
        }
    except Exception as e:
        logger.error(f"获取可恢复会话失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/history")
async def get_live_history(limit: int = Query(20, ge=1, le=100)) -> Dict[str, Any]:
    """获取直播历史记录"""
    try:
        logger.info(f"获取直播历史，limit={limit}")
        return {
            "success": True,
            "data": {
                "total": 0,
                "items": []
            }
        }
    except Exception as e:
        logger.error(f"获取直播历史失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/history/{session_id}")
async def get_session_history(session_id: str) -> Dict[str, Any]:
    """获取指定会话的历史记录"""
    try:
        logger.info(f"获取会话历史: {session_id}")
        return {
            "success": True,
            "data": {
                "session_id": session_id,
                "details": None
            }
        }
    except Exception as e:
        logger.error(f"获取会话历史失败: {e}")
        raise HTTPException(status_code=404, detail="会话不存在")


@router.delete("/history/{session_id}")
async def delete_session_history(session_id: str) -> Dict[str, Any]:
    """删除指定会话的历史记录"""
    try:
        logger.info(f"删除会话历史: {session_id}")
        return {
            "success": True,
            "message": "会话已删除"
        }
    except Exception as e:
        logger.error(f"删除会话失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/generate")
async def generate_report() -> Dict[str, Any]:
    """生成报告"""
    try:
        logger.info("生成报告")
        return {
            "success": True,
            "message": "报告生成中"
        }
    except Exception as e:
        logger.error(f"生成报告失败: {e}")
        raise HTTPException(status_code=500, detail=str(e))


@router.get("/review/{file_path:path}")
async def get_review_file(file_path: str) -> Dict[str, Any]:
    """获取评审文件"""
    try:
        logger.info(f"获取评审文件: {file_path}")
        return {
            "success": True,
            "data": {
                "path": file_path,
                "content": None
            }
        }
    except Exception as e:
        logger.error(f"获取评审文件失败: {e}")
        raise HTTPException(status_code=404, detail="文件不存在")
